docs/publishing/index.md

```diff
@@ -0,0 +1,123 @@
+
+Keep in mind that these suggestions are not specifically tuned for publishing to a CDN or other contexts where broader application-specific optimizations will not be applied to the code available therein.
+
+<style>
+.content__default ul {list-style:none;padding:0;}
+</style>
+
+### Do: 👍
+
+- ✅ [Do publish latest standard EcmaScript](#do-publish-latest-standard-ecmascript)
+- ✅ [Do publish standard es modules](#do-publish-standard-es-modules)
+- ✅ [Do include appropriate `package.json` entries](#do-include-appropriate-packagejson-entries)
+- ✅ [Do export element classes](#do-export-element-classes)
+- ✅ [Do export side effects separately](#do-export-side-effects-separately)
+- ✅ [Do keep side effects out of your index.js](#do-keep-side-effects-out-of-your-indexjs)
+- ✅ [Do import 3rd party node modules with "bare" import specifiers](#do-import-3rd-party-node-modules-with-bare-import-specifiers)
+- ✅ [Do include file extensions in import specifiers, always](#do-include-file-extensions-in-import-specifiers-always)
+- _Optional_: [use `type: "module"` when possible](#optional-use-type-module)
+- _Optional_: [include typings; e.g. `*.d.ts` files](#optional-include-typings)
+- _Optional_: [include source maps](#optional-include-source-maps)
+
+### Don't: 👎
+
+- ❌ [Do not bundle](#do-not-bundle)
+- ❌ [Do not minify](#do-not-minify)
+- ❌ [Do not optimize](#do-not-optimize)
+- ❌ [Do not use `.mjs` file extensions](#do-not-use-mjs-file-extensions)
+- ❌ [Do not import polyfills](#do-not-import-polyfills)
+
+## Do publish latest standard EcmaScript
+
+Contrary to some long-standing beliefs, an implementing project will always know more about the delivery context of the code it leverages than the developer of any of its dependencies. Taking this fact into account, the act of transpiling your code below a reasonably modern level (e.g. [TC39 Stage 4](https://github.com/tc39/proposals/blob/master/finished-proposals.md) and/or available cross-browser) is the best way that you can prepare your package to empower downstream projects to make their own decisions about the code they ship to their users.
+
+In some cases, you may choose to use newer specifications that may require special care when preparing for older browsers. In this case, transpiling down to the level you choose from above may be a good idea. If you choose not to do this, be sure to document this reality thoroughly so that your users are not left unawares.
```

Comment:
Maybe add an example of some kind of syntax/new spec that would need to be transpiled?

> If you choose not to do this, be sure to document this reality thoroughly so that your users are not left unawares.
We should not recommend people to ship newer, non-widely supported syntax in that case, right? Even if well documented?

Reply:
Good call, what the newest possible hotness to use as an example?

Good question, I think if I say "this uses X fancy API, be aware" that things would be cool. For instance, I probably wouldn't have transpiled `import.meta` in my code at any point in the past but would have wanted to document that it was used for users to be aware, but maybe that's wrong?

Reply:
> Good call, what the newest possible hotness to use as an example?

Maybe optional chaining is still a good example of this? 🤔 Or is it 'outdated' already? 😛

>  For instance, I probably wouldn't have transpiled import.meta in my code at any point in the past but would have wanted to document that it was used for users to be aware, but maybe that's wrong?

I think we discussed that stuff like this should be transpiled though, right? But I could be remembering it wrong

Reply:
I have altered this a good bit, maybe it's less troublesome now? Push coming in 20 or 30 minutes.